shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -334,7 +392,18 @@ class UCX(executorId: Int, usingWakeupFeature: Boolean = true) extends AutoClose
 
   private[ucx] def assignResponseTag(): Long = responseTag.incrementAndGet()
 
-  private def ucxWorkerAddress: ByteBuffer = worker.getAddress
+  private lazy val ucxAddress: ByteBuffer = if (rapidsConf.shuffleUcxUseSockaddr) {
+    val listenerAddress = listener.get.getAddress
+    val hostnameBytes = listenerAddress.getAddress.getAddress
+    val result = ByteBuffer.allocateDirect(4 + hostnameBytes.length)
+    result.putInt(listenerAddress.getPort)
+    result.put(hostnameBytes)
+    result.asReadOnlyBuffer()
+  } else {
+    worker.getAddress.asReadOnlyBuffer()
+  }
+
+  private def getUcxAddress: ByteBuffer = ucxAddress.duplicate().rewind()
```

Comment:
[The new buffer's capacity, limit, position, and mark values will be identical to those of this buffer.](https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html#asReadOnlyBuffer())

Reply:
Correct but: `The two buffers' position, limit, and mark values will be independent.` So I think you may want the `rewind()` in the lazy val, right before returning `asReadOnlyBuffer`. Since all we are doing is getting read only views of that, with independent markers, it will stay "rewound".